webapp/src/js/services/submit-form-by-sms.js

```diff
@@ -0,0 +1,57 @@
+angular.module('inboxServices').service('SubmitFormBySms',
+  function(
+    $log,
+    $q,
+    $window,
+    Form2Sms,
+    Settings
+  ) {
+    'use strict';
+    'ngInject';
+
+    return doc => {
+      if(!$window.medicmobile_android) {
+        $log.info('Not in android wrapper.');
+        return;
+      }
+
+      if(!$window.medicmobile_android.sms_available) {
+        $log.info('Android wrapper does not have SMS hooks.');
+        return;
+      }
+
+      if(!$window.medicmobile_android.sms_available()) {
+        $log.warn('Android wrapper does not have SMS enabled.  Check stacktrace to see why the SmsSender failed to initialise.');
+        return;
+      }
+
+      // Deliberately not returning this Promise, as we don't want to upset
+      // other workflows if the SMS fails.
+      $q.resolve()
```

Comment:
Wouldn't we want to return this so we can test it, but just document that you shouldn't hold onto the head when using it?

Reply:
:man_shrugging: I guess if it were needed for testing we could add a `return`.  I don't think the comment's very useful though, so I've removed it.